Annotation/Job.php

```diff
@@ -2,49 +2,54 @@
 
 namespace Padam87\CronBundle\Annotation;
 
+use Attribute;
+
 /**
  * @Annotation
  * @Target("CLASS")
  */
+#[Attribute(Attribute::TARGET_CLASS)]
 class Job
 {
-    /**
-     * @var string
-     */
-    public $minute = '*';
-
-    /**
-     * @var string
-     */
-    public $hour = '*';
-
-    /**
-     * @var string
-     */
-    public $day = '*';
-
-    /**
-     * @var string
-     */
-    public $month = '*';
-
-    /**
-     * @var string
-     */
-    public $dayOfWeek = '*';
-
-    /**
-     * @var string
-     */
-    public $commandLine;
-
-    /**
-     * @var string
-     */
-    public $group;
-
-    /**
-     * @var string
-     */
-    public $logFile;
+    public string $minute = '*';
+    public string $hour = '*';
+    public string $day = '*';
+    public string $month = '*';
+    public string $dayOfWeek = '*';
+
+    public ?string $group = null;
+
+    public ?string $logFile = null;
+
+    public ?string $commandLine = null;
+
+    public function __construct(
+        /** @var string|array */ $minute,
+        string $hour = null,
+        string $day = null,
+        string $month = null,
+        string $dayOfWeek = null,
+        string $group = null,
+        string $logFile = null,
+        ?string $commandLine = null
+    ) {
+        if (is_array($minute)) {
```

Comment:
I hate this, please don't do it this way.

Either do this on the caller side, or create a static method eg `fromAttribute`, and do it there.

Reply:
This is nessecary. By adding the __construct method (which is required for attributes, the annotations properties are processed will pass the __construct method, whereby all properties will be passed in the first argument as an array. 

See https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Validator/Constraints/Callback.php#L32 for the way Symfony handles this.

Using a static method will also not work, because the __construct is called from being the annotation code processing itself, so there is no way to trigger the static method from attributes. I have however created a method within the Job class with tne name process attributes to seperate this from the constructor.

See https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Validator/Constraint.php#L131

Reply:
I'm sorry, but I still don't get why that is relevant here.

Instead of `new Job(...$attribute->getArguments())` this should work just fine: `$attribute->newInstance()` no?

What is the reason of not doing it that way?

If there is a reason, please do it on the caller side.

Reply:
Ah, my bad, sorry. The Annotation doesn't have the `@NamedArgumentConstructor` yet, that is why its passing an array.

After a quick test it is working fine with that.

doctrine/annotations should be bumped to ^1.12 though.

Reply:
Hi Adam, thank you for your feedback. I've made some changes :)